src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,251 @@
+            return;
+        }
+
+        this._outputChannel.show();
+        await this.updateDistributedTracingSettingForDevices(deviceIds, iotHubConnectionString, updateType);
+
+        if (node instanceof DistributedTracingLabelNode) {
+            vscode.commands.executeCommand("azure-iot-toolkit.refresh", node);
+        } else if (node instanceof DistributedTracingSettingNode) {
+            vscode.commands.executeCommand("azure-iot-toolkit.refresh", node.parent);
+        } else {
+            vscode.commands.executeCommand("azure-iot-toolkit.refresh");
+        }
+    }
+
+    public async updateDistributedTracingSettingForDevices(deviceIds: string[], iotHubConnectionString: string, updateType: DistributedSettingUpdateType) {
+        let registry = iothub.Registry.fromConnectionString(iotHubConnectionString);
+
+        let mode: boolean = undefined;
+        let samplingRate: number = undefined;
+        let twin;
+
+        if (deviceIds.length === 1) {
+            twin = await Utility.getTwin(registry, deviceIds[0]);
+
+            if (twin.properties.desired[Constants.DISTRIBUTED_TWIN_NAME]) {
+                mode = Utility.parseDesiredSamplingMode(twin);
+                samplingRate = Utility.parseDesiredSamplingRate(twin);
+            }
+
+            if (updateType === DistributedSettingUpdateType.OnlySamplingRate) {
+                mode = undefined;
+            }
+
+            if (updateType === DistributedSettingUpdateType.OnlyMode) {
+                samplingRate = undefined;
+            }
+        }
+
+        if (updateType !== DistributedSettingUpdateType.OnlySamplingRate) {
+            let selectedItem: SamplingModeItem = await vscode.window.showQuickPick(
+                this.getSamplingModePickupItems(),
+                { placeHolder: "Select whether to enable/disable the distributed tracing...", ignoreFocusOut: true },
+            );
+            if (!selectedItem) {
+                return;
+            }
+            mode = selectedItem.distributedTracingEnabled;
+        }
+
+        if (updateType !== DistributedSettingUpdateType.OnlyMode) {
+            if (mode !== false) {
+                samplingRate = await this.promptForSamplingRate(`Enter sampling rate, integer within [0, 100]`, samplingRate);
+
+                if (samplingRate === undefined) {
+                    return;
+                }
+            }
+        }
+
+        TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingStartEvent);
+
+        await vscode.window.withProgress({
+            title: `Update Distributed Tracing Setting`,
+            location: vscode.ProgressLocation.Notification,
+        }, async () => {
+            try {
+                const result = await this.scheduleTwinUpdate(mode, samplingRate, iotHubConnectionString, deviceIds);
+                TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingDoneEvent, { Result: "Success" }, iotHubConnectionString);
```

Comment:
Also add DistributedSettingUpdateType to property. and device count?  And think about do we have other things to be added in BI?